R/loadFromMetaphlan.R

```diff
@@ -149,20 +149,27 @@ loadFromMetaphlan <- function(file, sample_meta = NULL, phy_tree = NULL, ...){
 }
 
 # Check that metaphlan file contains correct information
-.check_metaphlan <- function(data){
-    # Get rowdata columns
-    rowdata_columns <- data[ , 1:2]
-    # Get columns that go to assay
-    assay_columns <- data[ , 3:ncol(data)]
+.check_metaphlan <- function(x){
+    # Getting column indices of character columns
+    col_idx <- sapply(x, is.character)
+    # Get rowdata column indices
+    col_rowdata_idx <- col_idx[col_idx==TRUE]
```

Comment:
Does this work? To me it seems that col_idx is logical, then you identify cases that are TRUE; then you pick those cases from col_idx (same object) -> all resulting entries are TRUE by definition?

If you would pick index numbers then that should be given by `which(col_idx)` I think.

Reply:
I am sure that col_idx[col_idx==TRUE] returns logical vector, which(col_idx) return numeric vector. These are not the same. I guess which is what you need.

Reply:
Yes it does return logical vector, and I will switch it to `which`

Reply:
Sorry to re-iterate over this:
I have tested `which` with a case not having character columns but numeric columns and vice-versa. And it throws an error when called before reaching the if clause, making it non informative error to the user (not the error message supposed to be thrown by .check_metaphlan).
Initially in this function I did not have intention to subset any data for which the numeric indices would be needed. That's why I thought using boolean ones and operating on them at the if conditions later would be enough (which worked when tested line by line and by the files mentioned earlier at #306).
Sorry if I am inexperienced, but sub-setting with boolean values is not advised or unstable or?
Thanks